katsdpimager/templates/report.html.j2

```diff
@@ -0,0 +1,119 @@
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+    <title>Spectral imager report</title>
+    {{ resources | safe | indent(4) }}
+    {{ script | safe | indent(4) }}
+    <style>
+      {{ static["bootstrap-4.4.1.min.css"] | indent(6) }}
+    </style>
+  </head>
+  <body>
+    <div class="container-fluid">
+{% for target in targets %}
+      <div class="row">
+        <div class="col">
+          <h1>{{ target.name }}</h1>
+        </div>
+      </div>
+      <div class="row">
+        <div class="col-auto">
+          <div class="card">
+            <h5 class="card-header">Properties</h5>
+            <table class="table">
+              <tbody>
+                <tr>
+                  <td>Description</td>
+                  <td>{{ target.description }}</td>
+                </tr>
+                <tr>
+                  <td>Antennas</td>
+                  <td>{{ common.antennas | count }}</td>
+                </tr>
+                <tr>
+                  <td>Time on target</td>
+                  <td>{{ target.time_on_target | duration }}</td>
+                </tr>
+              </tbody>
+            </table>
+          </div>
+        </div>
+        <div class="col-auto">
+          <div class="card">
+            <h5 class="card-header">Channels</h5>
+            <table class="table">
+              <!-- <colgroup><col style="width: 10em"><col></colgroup> -->
+              <tbody>
+                <tr>
+                  <td>Successful</td>
+                  <td>{{ target.status|select("equalto", "complete")|list|count }}</td>
+                </tr>
+                <tr>
+                  <td>Flagged</td>
+                  <td>{{ target.status|select("equalto", "no-data")|list|count }}</td>
+                </tr>
+                <tr>
+                  <td>Masked</td>
+                  <td>{{ target.status|select("equalto", "masked")|list|count }}</td>
+                </tr>
+                <tr>
+                  <td>Failed</td>
+                  <td>{{ target.status|select("equalto", "failed")|list|count }}</td>
+                </tr>
+                <tr>
+                  <td>Total</td>
+                  <td>{{ common.channels }}</td>
+                </tr>
+              </tbody>
+            </table>
+          </div>
+        </div>
+      </div>
+      <div class="row">
+        <div class="col">
+          <div class="card">
+            <h5 class="card-header">Processing status</h5>
+            {{ target.plots['status'] | safe | indent(12) }}
+          </div>
+          <div class="card">
+            <h5 class="card-header">Flux density</h5>
+            {{ target.plots['flux'] | safe | indent(12) }}
+            <p class="card-body">
+              <dl>
+                <dt>Peak</dt>
+                <dd>Maximum flux density of any pixel in the final image, after
+                primary beam correction.</dd>
```

Comment:
Two niggles:
1. The units of the peak in an image are not strictly 'Flux Density' (Nor are they 'Flux' as in the y-axis label of the plot). The value of a pixel (and the pixel noise) is in units of flux density per solid angle which is more commonly referred to in astronomy as a 'surface brightness'. However I wouldn't use that since it is a misleading term in this context (in astronomy it's normally only used as a quantity describing objects that are smooth and diffuse and have no defined 'peak'). It has always been a problem coming up with the correct term to use for a pixel value or noise in an image (I'm guilty of calling it flux density or 'peak flux' myself much of the time). For SUMSS we chose 'Amplitude' or 'Peak Amplitude' which got past our (very pedantic and knowledgeable) referee. Will that work for you - I'm open to any other suggestions (actually Briggs might have something to say here?).

2. I think that you should plot the peak both before and after PB correction here. The 'before PB' flux gives a 0th order estimate of dynamic range and is useful. The different slopes of the 'corrected' and 'uncorrected' flux as a function of frequency will also give a quick visual indication of how far off the phase center a source is for a more 'experienced' viewer of the plot.

Reply:
I guess the right term depends on whether the source is resolved or not. I can probably get away with just labelling it 'Peak'.

I can take a look at also looking at the uncorrected peak. Also to be able to see anything as a function of frequency assumes that the emission has enough bandwidth to see a slope but not enough to get continuum subtracted away...

Reply:
>I guess the right term depends on whether the source is resolved or not. I can probably get away with just labelling it 'Peak'.

'Peak' will work. Just not 'flux density' which has units of Jy not Jy/beam (even though these are equal for a point source they are still different things).

>I can take a look at also looking at the uncorrected peak. Also to be able to see anything as a function of frequency assumes that the emission has enough bandwidth to see a slope but not enough to get continuum subtracted away...

Yes. I was thinking too 'continuum'-ly here (pardon me). I still think it's perhaps worth plotting both (The uncorrected peak will still give a very simplistic DR estimate.).

Reply:
It looks like uncorrected peak going to be problematic, because I never actually create a combined image prior to primary beam correction. The primary beam correction is applied before the restoring beam. The closest I could get is peak in the uncorrected model image - but I'm not convinced that can be usefully compared to peak pixel in the final image. Do you think it's critical or should I just leave it for now?

Reply:
I see. I just worry a bit about the case where the corrected peak of the noise at the edge of an image is higher then the peak at the center - thus you will be missing peaks of 'detections' fainter than the noise at the edge that are in the middle of the image.

Reply:
True, I guess for most of the band the "peak" is fairly meaningless because it'll be noise at whatever radius we truncate the image.

How about the following:
1. Find the position of the largest clean component (assuming there was any cleaning and that it wasn't all noise to start with).
2. Look up that pixel in the final image.
3. Multiply it by the primary beam for that pixel to get an uncorrected value.

Reply:
Yup. That should work.

Reply:
Ok, will try that on Monday. Or a possible variation would be to take the largest pixel value in the final image but restrict it to pixels with clean components (hopefully a bit less sensitive to how much of the pixel came from cleaned versus residual, as well as issues around flux being split across 2x2 model pixels in an uneven way).

In either case I guess there is still some risk that a false detection near the edge of the beam (because even at 5 sigma there are likely to be a few pixels) will look like a peak. I suppose we'll just need to see how it pans out.

Reply:
Why not restrict the peak detection to inside the HPBW or something similar ? It is not of course a truly accurate measure, but it will be pretty consistent.

Reply:
I don't think the cutoff radius really matters - whatever you choose, if there is no real emission then your peak is just going to be some (fairly random) multiple of the RMS noise divided by the beam power at the cutoff radius. What I think might help though is limiting the search to pixels where the model image is at least, say, 10 sigma prior to primary beam correction, which should ensure that only real emission is considered, at the expense of not picking up faint emission.

Reply:
It looks like 10 sigma may be too high: it's definitely missing real emission in a short DEEP2 offset observation, which is at about 7-8 sigma (haven't measured exactly). I can reduce to 7.5 sigma without creating many false detections (expected about 0.02 across all channels), but eventually we may need something smarter that looks at spatial or temporal structure.